src/components/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,238 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import * as d3 from 'd3';
+import { IBarChartProps, IBarChartData } from './BarChart.props';
+import './BarChart.scss';
+
+enum Update {
+    Data = 1,
+    Dimension
+};
+
+export class BarChart extends React.Component<IBarChartProps, null> {
+
+    public static defaultProps = {
+        id: '',
+        barColor: '#889ac4',
+        hovColor: '#b8c7e8',
+        data: [],
+        tipText: (data: IBarChartData) => ('Frequency of ' + data.argument + ' is : ' + data.frequency),
+        xAxisFormat: () => null,
+        width: 500,
+        height: 300,
+        minWidth: 300,
+        maxWidth: 1000,
+        isResponsive: false,
+        onClick: () => { return; }
+    };
+
+    public refs: {
+        [argument: string]: (Element);
+        container: (HTMLInputElement);
+    };
+
+    private margin = { top: 50, right: 50, bottom: 50, left: 50 };

```

Comment:
These props just hold references to selections placing them in state will just means that there could be some unnecessary re-rendering